enhancements/agent-installer/automated-workflow.md

```diff
@@ -0,0 +1,491 @@
+  and to tools such as the zero-touch provisioning (ZTP) system in
+  multi-cluster engine (MCE).
+- On-premises deployments do not always have DHCP or other remote
+  services available to configure host networking. They also
+  frequently use more complex network configurations such as bonded
+  NICs and tagged VLANs. We therefore need to support configuration of
+  per-host network settings while installing.
+- On-premises deployments, especially on bare metal, may need to take
+  extra steps to fully utilize storage inside each host or visible to
+  a host from a SAN. We therefore need to support configuration of
+  per-host storage, including fibre channel settings such as
+  multipath.
+- We want to retain the OpenShift installer's ability to customize a
+  deployment with arbitrary manifests.
+
+### Non-Goals
+
+- A user-driven workflow will be described in a separate enhancement.
+- A user interface for collecting complex inputs such as per-host
+  network settings before running the installation automatically is
+  left to a future enhancement.
+- Changes to the release image mirroring process are out of scope for
+  this enhancement.
+- Eliminating the requirement for a separate image registry to host
+  the mirrored OpenShift image is left to a future enhancement.
+- Describing a mechanism to add and remove nodes from the cluster after
+  it is installed is left to a future enhancement.
+- Generating image formats other than ISO is left to a future
+  enhancement.
+- Booting hosts via PXE is left to a future enhancement.
+- Creating minimal ISO images with a separate rootfs is left to a
+  future enhancement.
+- We are not currently considering replacing the IPI installer or
+  assisted installer SaaS.
+- Describing an end-to-end solution for users without their own
+  orchestration tool is left to a future enhancement.
+- Collecting telemetry data for connected deployments is left to a
+  future enhancement.
+- Details of securing the communication between agents and services to
+  ensure only desired hosts join the cluster are left to a future
+  enhancement.
+- Details of a workflow for deploying in settings with DHCP assignment
+  of IP addresses is left to a future enhancement.
+
+## Proposal
+
+We propose to extend the `openshift-install` binary with a new
+sub-command, `create image` for generating a bootable image containing
+all of the information needed to deploy a cluster.  A command line
+tool avoids issues with prerequisites for deploying a long-running
+service, and can be run on a low-resource host such as a laptop or
+even on a host at a different site before deployment begins.
+
+Most enterprise grade hardware includes a management controller that
+can present an ISO image to a host virtually, allowing the host to
+boot from the image remotely. Hypervisors such as kvm and vSphere also
+support booting from ISOs. Therefore, we will concentrate on that
+format and approach first. Building minimal ISOs with separate root
```

Comment:
Minimal ISO is very important for virtual media over slower networks but again, we can leave that out of scope for now.